go-garbage-collector/README.md

```diff
@@ -0,0 +1,73 @@
+# Instrumenting the Go Garbage Collector with uprobes
+
+Instrument the internals of the Golang garbage collector to visualize its behavior. [Dumpster-diving the Go Garbage Collector](https://blog.px.dev/go-garbage-collector) is the accompanying blog post for this demo.
+
+<img src=".readme_assets/sweep.png" alt="Pages swept over time by the Go garbage collector">
+
+## What is this demo?
+
+This demo provides instructions for using [Pixie](https://github.com/pixie-io/pixie) to instrument the Go garabge collector. It provides an example Go 1.16 application as well as the specification for the probes in [PxL](https://docs.px.dev/reference/pxl/).
+
+<img src=".readme_assets/gc_events.png" alt="Allocate and sweep events in the Go garbage collector">
+
+You can use this demo to visualize
+- GC events and their latencies and timestamps
+- Pages allocated, marked, and swept in the Go runtime over time
+- The amount of assist work done by the runtime over time
+- The trigger ratio for the garbage collector over time
+
+## Prereqquisites
+
+* If you don't already have one, set up a [Kubernetes cluster](https://docs.px.dev/installing-pixie/setting-up-k8s/)
+* [Install Pixie](https://docs.px.dev/installing-pixie/install-guides/) to your Kubernetes cluster
+
+## Usage
+
+Notes before beginning: 
+- This demo uses the [Dynamic Go Logging](https://docs.px.dev/tutorials/custom-data/dynamic-go-logging/) feature of Pixie. This feature is in still in alpha.
+- This demo uses an application running Go 1.16. It will not work on other versions of Go.
+
+1. Deploy the test application.
+
+```
+# Will be created in the `default` namespace
+kubectl apply -f demo-app.yaml
+```
+
+2. Get the name of the pod you just deployed, starting with `gc-service`.
+
+```
+kubectl get pods
+```
+
+3. `git clone` this repo.
+
+4. In a text editor, open the `.pxl` files in the `pxl_scripts/uprobe_scripts` and `pxl_scripts/visualization` directories. Fill in the `pod` variable at the top of each of the scripts to be `default/gc-service-<rest of pod name>`.
+
+5. Open the Pixie UI and navigate to the [Scratch Pad](https://docs.px.dev/using-pixie/using-live-ui/#write-your-own-pxl-scripts-use-the-scratch-pad).
+
+6. Follow the [Scratch Pad](https://docs.px.dev/using-pixie/using-live-ui/#write-your-own-pxl-scripts-use-the-scratch-pad) instructions to run the canary script at `pxl_scripts/uprobe_scripts/_canary_probe.pxl`. You do not need to provide a `vis.json` file.
```

Comment:
Elaborate on what success looks like: "You should see several rows of data in the output table." (I'm not actually sure what success is because my canary probe failed. :/)

Reply:
Or is success an empty table....but the probe passes?